framework/validators/AtLeastValidator.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\validators;
+
+use Yii;
+use yii\base\InvalidConfigException;
+
+/**
+ * AtLeastValidator checks if at least one of many attributes are filled.
+ *
+ * In the following example, the `attr1`, `attr2` and `attr3` attributes will
+ * be verified. If at least one of them are not filled, all will receive an error:
+ *
+ * ~~~[php]
+ *      // in rules()
+ *      return [
+ *          ['attr1', 'atLeast', 'with' => ['attr2', 'attr3']],
+ *      ];
+ * ~~~
+ *
+ * In the following example, the `attr1`, `attr2` and `attr3` attributes will
+ * be verified. If at least two of them are not filled, `attr1` and `attr3` will
+ * receive an error:
+ *
+ * ~~~[php]
+ *      // in rules()
+ *      return [
+ *          ['attr1', 'atLeast', 'with' => ['attr2', 'attr3'], 'min' => 2, 'errorIn' => ['attr1', 'attr3']],
+ *      ];
+ * ~~~
+ *
+ * Important: Only one attribute should be specified in first param of the rule.
+ * The aditional ones must go into [[with]] param.
+ *
+ * The attributes that will receive the error message can be specified into [[errorIn]]
+ * param. If none is specified, all envolved attributes will receive the error.
+ *
+ * @author Sidney Lins <slinstj@gmail.com>
+ * @since 2.0.4
+ */
+class AtLeastValidator extends Validator
```

Comment:
I'm not really happy with the name, "at least" is missing something. at least... what? Alternatives:

- RequireFrom
- RequireOne
- RequireAny
- Alternative
- AtLeastOne
- ...